### PR TITLE
[qt5] Remove dependency qt5 feature mysqlplugin when building x86

### PR DESCRIPTION
**Pull request by JackBoosY**
Since MySQL upstream has stopped supporting x86, when installing `qt5:x86-windows`, the error will be reported:
```
CMake Error at scripts/cmake/vcpkg_fail_port_install.cmake:97 (message):
  Architecture 'x86' not supported by libmysql!
```
Fix this.

Related: #15170.

### CODE REVIEW

**Comment by JackBoosY** [comment]
Ping @dan-shaw for review and merge this PR.

File: ports/qt5/CONTROL [hunk 1]
```diff
@@ -1,9 +1,9 @@
 Source: qt5
 Version: 5.15.1
-Port-Version: 1
+Port-Version: 2
 Homepage: https://www.qt.io/
 Description: Qt5 Application Framework
-Build-Depends: qt5-base[core,mysqlplugin]
+Build-Depends: qt5-base
```
Comment (JackBoosY) [inline ports/qt5/CONTROL:6]:
```suggestion
Build-Depends: qt5-base[core,mysqlplugin] (!x86), qt5-base (x86)
```

**Review by NancyLi1013 — APPROVED** [review]

**Comment by vicroms** [comment]
Thanks @JackBoosY 

I also believe that it shouldn't be a default feature.

**Comment by dan-shaw** [comment]
After thinking about it, I agree with Victor, can we remove it from the default features instead?

File: ports/qt5/CONTROL [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Source: qt5
 Version: 5.15.1
-Port-Version: 1
+Port-Version: 2
 Homepage: https://www.qt.io/
 Description: Qt5 Application Framework
-Build-Depends: qt5-base[core,mysqlplugin]
+Build-Depends: qt5-base[core,mysqlplugin] (!x86), qt5-base (x86)
```
Comment (JackBoosY) [inline ports/qt5/CONTROL:6]:
```suggestion
Build-Depends: qt5-base[core]
```

**Comment by JackBoosY** [comment]
@dan-shaw Done.

**Pull request merged**